pkg/sql/show_cluster_setting.go

```diff
@@ -156,10 +156,14 @@ func (p *planner) ShowClusterSetting(
 	ctx context.Context, n *tree.ShowClusterSetting,
 ) (planNode, error) {
 	name := settings.SettingName(strings.ToLower(n.Name))
-	setting, ok := settings.LookupForLocalAccess(name, p.ExecCfg().Codec.ForSystemTenant())
+	setting, ok, nameStatus := settings.LookupForLocalAccess(name, p.ExecCfg().Codec.ForSystemTenant())
 	if !ok {
 		return nil, errors.Errorf("unknown setting: %q", name)
 	}
+	if nameStatus != settings.NameActive {
+		p.BufferClientNotice(ctx, pgnotice.Newf("the name %q is deprecated; use %q instead", name, setting.Name()))
```

Comment:
[nit] I suppose you could move this error message construction to some shared functions so they don't get out of sync.